### PR TITLE
CP-24868: Change vm_bios_string_keys to Map(String, String) format

### PR DESCRIPTION
**Pull request by sharady**

### CODE REVIEW

File: ocaml/xapi/xapi_vm_helpers.ml [hunk 1]
```diff
@@ -933,10 +933,13 @@ let get_SRs_required_for_recovery ~__context ~self ~session_to =
...
 (* BIOS strings *)
-let assert_valid_bios_strings ~__context ~bios_strings =
+let assert_valid_bios_strings ~__context ~value =
+  (* Validate BIOS string keys *)
   (* Validate size of value provided is within bios_string_limit_size and not empty *)
   (* Validate value chars are printable ASCII characters *)
-  bios_strings |> List.iter (fun (k, v) ->
+  value |> List.iter (fun (k, v) ->
+    if not (List.mem k Xapi_globs.settable_vm_bios_string_keys) then
+      raise (Api_errors.Server_error(Api_errors.invalid_value, [k]));
```
Comment (robhoes) [inline ocaml/xapi/xapi_vm_helpers.ml]: The `invalid_value` error is defined as `error Api_errors.invalid_value ["field"; "value"]`, so it needs two arguments.

File: ocaml/xapi/records.ml [hunk 2]
```diff
@@ -1003,8 +1003,9 @@ let vm_record rpc session_id vm =
         ~get:(fun () -> Record_util.s2sm_to_string "; " (x ()).API.vM_bios_strings)
         ~get_map:(fun () -> (x ()).API.vM_bios_strings)
         ~set_map:(fun x ->
-          let bios_strings = List.map (fun (k, v) -> Record_util.string_to_vm_bios_key k, v) x in
-          Client.VM.set_bios_strings rpc session_id vm bios_strings)();
+          List.iter (fun (k, v) -> if not (List.mem k Xapi_globs.settable_vm_bios_string_keys) then
+            raise (Record_util.Record_failure ("Unknown key '"^k^"': expecting " ^ (String.concat ", " Xapi_globs.settable_vm_bios_string_keys)))) x;
```
[the remark this answers is not in the document]
Comment (sharady) [reply ocaml/xapi/records.ml]: From CLI, there is not other way to reflect the expected keys, until we add them in API error.

**Comment by sharady** [comment]
@robhoes Thanks for the review, Please have a look once more.

File: ocaml/xapi/records.ml (hunk 2 above)
[the remark this answers is not in the document]
Comment (robhoes) [reply ocaml/xapi/records.ml]: This isn't very nice, but I'll let it go for now.

**Pull request merged**